### PR TITLE
Filter unsupported artifact types from analyzer sbom regardless of ANCHORE_ENABLE_PACKAGE_FILTERING configuration.

### PR DESCRIPTION
**Pull request by Vijay-P**
Move filtering of unsupported types from filter_artifacts to convert_syft_to_engine.

This change will prevent unsupported artifact types from causing a KeyError to be raised when package filtering is disabled.

Behavior before change:
```
[service:enterprise-analyzer] 2021-07-12 12:17:51+0000 [-] Traceback (most recent call last):
[service:enterprise-analyzer] 2021-07-12 12:17:51+0000 [-] File "/usr/local/lib/python3.8/site-packages/anchore_engine/clients/localanchore_standalone.py", line 1112, in analyze_image
[service:enterprise-analyzer] 2021-07-12 12:17:51+0000 [-] analyzer_report = run_anchore_analyzers(
[service:enterprise-analyzer] 2021-07-12 12:17:51+0000 [-] File "/usr/local/lib/python3.8/site-packages/anchore_engine/clients/localanchore_standalone.py", line 909, in run_anchore_analyzers
[service:enterprise-analyzer] 2021-07-12 12:17:51+0000 [-] analyzer_report = analyzer_manager.run(
[service:enterprise-analyzer] 2021-07-12 12:17:51+0000 [-] File "/usr/local/lib/python3.8/site-packages/anchore_engine/analyzers/manager.py", line 26, in run
[service:enterprise-analyzer] 2021-07-12 12:17:51+0000 [-] _run_syft(
[service:enterprise-analyzer] 2021-07-12 12:17:51+0000 [-] File "/usr/local/lib/python3.8/site-packages/anchore_engine/analyzers/manager.py", line 117, in _run_syft
[service:enterprise-analyzer] 2021-07-12 12:17:51+0000 [-] results = syft.catalog_image(
[service:enterprise-analyzer] 2021-07-12 12:17:51+0000 [-] File "/usr/local/lib/python3.8/site-packages/anchore_engine/analyzers/syft/_init_.py", line 43, in catalog_image
[service:enterprise-analyzer] 2021-07-12 12:17:51+0000 [-] return convert_syft_to_engine(all_results, package_filtering_enabled)
[service:enterprise-analyzer] 2021-07-12 12:17:51+0000 [-] File "/usr/local/lib/python3.8/site-packages/anchore_engine/analyzers/syft/_init_.py", line 80, in convert_syft_to_engine
[service:enterprise-analyzer] 2021-07-12 12:17:51+0000 [-] handler = modules_by_artifact_type[artifact["type"]]
[service:enterprise-analyzer] 2021-07-12 12:17:51+0000 [-] KeyError: 'go-module'
[service:enterprise-analyzer] 2021-07-12 12:17:51+0000 [-]
[service:enterprise-analyzer] 2021-07-12 12:17:51+0000 [-] During handling of the above exception, another exception occurred:
[service:enterprise-analyzer] 2021-07-12 12:17:51+0000 [-]
[service:enterprise-analyzer] 2021-07-12 12:17:51+0000 [-] Traceback (most recent call last):
[service:enterprise-analyzer] 2021-07-12 12:17:51+0000 [-] File "/usr/local/lib/python3.8/site-packages/anchore_engine/services/analyzer/analysis.py", line 340, in process_analyzer_job
[service:enterprise-analyzer] 2021-07-12 12:17:51+0000 [-] image_data = perform_analyze(
[service:enterprise-analyzer] 2021-07-12 12:17:51+0000 [-] File "/usr/local/lib/python3.8/site-packages/anchore_engine/services/analyzer/analysis.py", line 194, in perform_analyze
[service:enterprise-analyzer] 2021-07-12 12:17:51+0000 [-] analyzed_image_report, manifest_raw = localanchore_standalone.analyze_image(
[service:enterprise-analyzer] 2021-07-12 12:17:51+0000 [-] File "/usr/local/lib/python3.8/site-packages/anchore_engine/clients/localanchore_standalone.py", line 1141, in analyze_image
[service:enterprise-analyzer] 2021-07-12 12:17:51+0000 [-] raise AnalysisError(
[service:enterprise-analyzer] 2021-07-12 12:17:51+0000 [-] anchore_engine.clients.localanchore_standalone.AnalysisError: failed to download, unpack, analyze, and generate image export (<REDACTED>@sha256:ed7dde042d73fef24fe61c1637fdf7cbdf6a75c68372bb884651a113cb7898b1) - exception: 'go-module'
[service:enterprise-analyzer] 2021-07-12 12:17:51+0000 [-] [Thread-513546] [anchore_engine.services.analyzer.analysis/process_analyzer_job()] [ERROR] problem analyzing image - exception: failed to download, unpack, analyze, and generate image export (<REDACTED>@sha256:ed7dde042d73fef24fe61c1637fdf7cbdf6a75c68372bb884651a113cb7898b1) - exception: 'go-module'
```

Behavior after change:
```
analyzer_1       | [service:worker] 2021-07-21 17:12:24+0000 [-] [Thread-20] [anchore_engine.analyzers.syft/convert_syft_to_engine()] [INFO] filtering owned packages
analyzer_1       | [service:worker] 2021-07-21 17:12:24+0000 [-] [Thread-20] [anchore_engine.analyzers.syft/convert_syft_to_engine()] [WARN] Handler for artifact type go-module not available. Skipping package <REDACTED>.
analyzer_1       | [service:worker] 2021-07-21 17:12:24+0000 [-] [Thread-20] [anchore_engine.analyzers.syft/convert_syft_to_engine()] [WARN] Handler for artifact type go-module not available. Skipping package <REDACTED>.
analyzer_1       | [service:worker] 2021-07-21 17:12:24+0000 [-] [Thread-20] [anchore_engine.analyzers.syft/convert_syft_to_engine()] [WARN] Handler for artifact type go-module not available. Skipping package <REDACTED>.
analyzer_1       | [service:worker] 2021-07-21 17:12:24+0000 [-] [Thread-20] [anchore_engine.analyzers.syft/convert_syft_to_engine()] [WARN] Handler for artifact type go-module not available. Skipping package <REDACTED>.
analyzer_1       | [service:worker] 2021-07-21 17:12:24+0000 [-] [Thread-20] [anchore_engine.analyzers.syft/convert_syft_to_engine()] [WARN] Handler for artifact type go-module not available. Skipping package <REDACTED>.
analyzer_1       | [service:worker] 2021-07-21 17:12:24+0000 [-] [Thread-20] [anchore_engine.analyzers.syft/convert_syft_to_engine()] [WARN] Handler for artifact type go-module not available. Skipping package github.com/golang/protobuf.
analyzer_1       | [service:worker] 2021-07-21 17:12:24+0000 [-] [Thread-20] [anchore_engine.analyzers.syft/convert_syft_to_engine()] [WARN] Handler for artifact type go-module not available. Skipping package github.com/pkg/errors.
analyzer_1       | [service:worker] 2021-07-21 17:12:24+0000 [-] [Thread-20] [anchore_engine.analyzers.syft/convert_syft_to_engine()] [WARN] Handler for artifact type go-module not available. Skipping package github.com/spf13/viper.
analyzer_1       | [service:worker] 2021-07-21 17:12:24+0000 [-] [Thread-20] [anchore_engine.analyzers.syft/convert_syft_to_engine()] [WARN] Handler for artifact type go-module not available. Skipping package github.com/stretchr/testify.
analyzer_1       | [service:worker] 2021-07-21 17:12:24+0000 [-] [Thread-20] [anchore_engine.analyzers.syft/convert_syft_to_engine()] [WARN] Handler for artifact type go-module not available. Skipping package <REDACTED>.
analyzer_1       | [service:worker] 2021-07-21 17:12:24+0000 [-] [Thread-20] [anchore_engine.analyzers.syft/convert_syft_to_engine()] [WARN] Handler for artifact type go-module not available. Skipping package <REDACTED>.
analyzer_1       | [service:worker] 2021-07-21 17:12:24+0000 [-] [Thread-20] [anchore_engine.analyzers.syft/convert_syft_to_engine()] [WARN] Handler for artifact type go-module not available. Skipping package <REDACTED>.
analyzer_1       | [service:worker] 2021-07-21 17:12:24+0000 [-] [Thread-20] [anchore_engine.analyzers.syft/convert_syft_to_engine()] [WARN] Handler for artifact type go-module not available. Skipping package <REDACTED>.
analyzer_1       | [service:worker] 2021-07-21 17:12:24+0000 [-] [Thread-20] [anchore_engine.analyzers.syft/convert_syft_to_engine()] [WARN] Handler for artifact type go-module not available. Skipping package <REDACTED>.
analyzer_1       | [service:worker] 2021-07-21 17:12:24+0000 [-] [Thread-20] [anchore_engine.analyzers.syft/convert_syft_to_engine()] [WARN] Handler for artifact type go-module not available. Skipping package golang.org/x/net.
analyzer_1       | [service:worker] 2021-07-21 17:12:25+0000 [-] [Thread-20] [anchore_engine.services.analyzer.analysis/perform_analyze()] [INFO] performing analysis on image complete: 192.168.0.1:5000/customer-debug@sha256:ed7dde042d73fef24fe61c1637fdf7cbdf6a75c68372bb884651a113cb7898b1
```

### CODE REVIEW

File: anchore_engine/analyzers/syft/__init__.py [hunk 1]
```diff
@@ -19,11 +21,6 @@ def filter_fn(relationship):
 
 def filter_artifacts(artifacts, relationships):
     def filter_fn(artifact):
-        # syft may do more work than what is supported in engine, ensure we only include artifacts
-        # of select package types.
-        if artifact["type"] not in modules_by_artifact_type:
```
Comment (dspalmer99) [inline anchore_engine/analyzers/syft/__init__.py:24]: Does any of the upstream logic that calls this function need this check?

File: tests/unit/anchore_engine/analyzers/syft/test_syft.py [hunk 2]
```diff
@@ -145,3 +129,28 @@ def test_does_not_filter_artifact_by_type(self, pkg_type):
...
+class TestConvertSyftToEngine:
+    @pytest.mark.parametrize(
+        "enable_package_filtering",
+        [False, True],
+    )
+    def test_filter_artifact_by_type(self, test_sbom, enable_package_filtering):
+        findings = convert_syft_to_engine(test_sbom, enable_package_filtering)
+        for pkg_list in findings["package_list"]:
+            assert (
+                "TEST-PACKAGE" not in findings["package_list"][pkg_list]["base"].keys()
```
Comment (dspalmer99) [inline tests/unit/anchore_engine/analyzers/syft/test_syft.py:155]: Could you add some checks for packages that are expected to be found?

File: anchore_engine/analyzers/syft/__init__.py (hunk 1 above)
Comment (Vijay-P) [reply anchore_engine/analyzers/syft/__init__.py:24]: nothing else calls the function

File: tests/unit/anchore_engine/analyzers/syft/test_syft.py (hunk 2 above)
Comment (Vijay-P) [reply tests/unit/anchore_engine/analyzers/syft/test_syft.py]: done

File: anchore_engine/analyzers/syft/__init__.py (hunk 1 above)
Comment (dspalmer99) [reply anchore_engine/analyzers/syft/__init__.py:24]: Ack, thanks Vijay

**Review by dspalmer99 — APPROVED** [review]
LGTM

**Review by dakaneye — APPROVED** [review]
lgtm, and like the tests that were added

**Pull request merged**